File: Matrix.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void add();
void subtraction();
void multiply();

int arr[100][100],arr2[100][100], arr3[100][100]; 
int i, j, k;
int choice;
int row, column;

int main(){

    clock_t start = clock();

    srand(time(0));

    printf("Enter the size of row: ");
    scanf("%d", &row);

    printf("Enter the size of column: ");
    scanf("%d", &column);


    //Enter the first matrix1

    for(i=0; i<row; i++){
        for ( j = 0; j < column; j++)
        {
            arr[i][j] = (rand()%100);
        }
        
    }

    //Enter the second matrix

    for(i=0; i<row; i++){
        for ( j = 0; j < column; j++)
        {
            arr2[i][j] = (rand()%100);
        }
        
    }

    printf("Elements of the array one: \n");

    for ( i = 0; i < row; i++)
    {
        printf("\n");
        for ( j = 0; j < column; j++)
        {
            printf("%d\t", arr[i][j]);
        }
        
    }

    printf("\n");

    printf("Elements of the array two: \n");

    for ( i = 0; i < row; i++)
    {
        printf("\n");
        for ( j = 0; j < column; j++)
        {
            printf("%d\t", arr2[i][j]);
        }
        
    }
    
    printf("\n");

    do
    {
        printf("--------- ENTER A NUMBER TO SELECT THE OPERATION ----------- \n");
        printf("1 :     MATRIX ADDITION \n");
        printf("2 :     MATRIX SUBTRACTION \n");
        printf("3 :     MATRIX MULTIPLICATION\n");
        printf("4 :     EXIT \n");

        scanf("%d", &choice);

        switch (choice)
        {
        case 1:
            add();
            break;

        case 2:
            subtraction();
            break;

        case 3:
            multiply();
            break;

        case 4:
            break;
        
        default:
            printf("INVALID OPTION. TRY AGAIN!!!");
            break;
        }
    } while (choice != 4);
    
    printf("\n"); 

    clock_t end = clock();
    printf("Elapsed: %f seconds\n", (double)(end - start) / CLOCKS_PER_SEC); 
}

void multiply(){

    printf("Multiplication of the arrays:\n");

    for ( i = 0; i < row; i++)
    {
        for ( j = 0; j < column; j++)
        {
            arr3[i][j] = 0;
            for ( k = 0; k < row; k++)
            {
                arr3[i][j] += arr[i][k]*arr2[k][j];
            }
        }
    }

    printf("Result: \n");

    for ( i = 0; i < row; i++)
    {
        printf("\n");
        for ( j = 0; j < row; j++)
        {
            printf("%d\t", arr3[i][j]);
        }
    }

    printf("\n");
}

void add(){

    printf("Addition of the arrays:\n");

    for ( i = 0; i < row; i++)
    {
        for ( j = 0; j < column; j++)
        {
            arr3[i][j] = arr[i][j] + arr2[i][j];    
        }
    }

    printf("Result: \n");

    for ( i = 0; i < column; i++)
    {
        printf("\n");
        for ( j = 0; j < row; j++)
        {
            printf("%d\t", arr3[i][j]);
        }
    }

    printf("\n");
}

void subtraction(){

    printf("Subtraction of the Matrix:\n");

    for ( i = 0; i < row; i++)
    {
        for ( j = 0; j < column; j++)
        {
            arr3[i][j] = arr[i][j] - arr2[i][j];
        }
    }

    printf("Result: \n");

    for ( i = 0; i < row; i++)
    {
        printf("\n");
        for ( j = 0; j < column; j++)
        {
            printf("%d\t", arr3[i][j]);
        } 
    }

    printf("\n");
}

File: Matrix.java
import java.util.Random;
import java.util.Scanner;

public class Matrix {

    static int arr1[][] = new int[100][100];
    static int arr2[][] = new int[100][100];
    static int arr3[][] = new int[100][100];
    static int row;
    static int column;

    public static void main(String[] args) {

        final long starttime = System.nanoTime();

        Random rand = new Random();

        Scanner input = new Scanner(System.in);

        System.out.println("Enter the number of rows: ");
        row = input.nextInt();

        System.out.println("Enter the number of column: ");
        column = input.nextInt();

        //System.out.println("Enter values for matrix 1");

        for (int i = 0; i < row; i++) {
            for (int j = 0; j < column; j++) {
                //System.out.print("[" + i + "] [" + j + "] : ");
                arr1[i][j] = rand.nextInt(100);
            }
        }

        //System.out.println("Enter values for matrix 2");

        for (int i = 0; i < row; i++) {
            for (int j = 0; j < column; j++) {
                //System.out.print("[" + i + "] [" + j + "] : ");
                arr2[i][j] = rand.nextInt(100);
            }
        }

        System.out.println("Values for matrix 1");

        for(int i = 0; i < row; i++){
            System.out.println("\n");
            for(int j = 0; j < column; j++){
                System.out.print(arr1[i][j]+"\t");
            }
        }

        System.out.println("\n");

        System.out.println("Values for matrix 2");

        for(int i = 0; i < row; i++){
            System.out.println("\n");
            for(int j = 0; j < column; j++){
                System.out.print(arr1[i][j]+"\t");
            }
        }

        int option;

        do{

            System.out.println("====================== SELECT AN OPTION ====================");
            System.out.println("1 : \t ADDITION OPERATION");
            System.out.println("2 : \t SUBTRACTION OPERATION");
            System.out.println("3 : \t MULTIPLICATION OPERATION");
            System.out.println("4 : \t EXIT OPERATION");

            System.out.print("Enter Your Option");
            option = input.nextInt();

            switch (option) {
                case 1:
                    addition();
                    break;

                case 2:
                    subtraction();
                    break;

                case 3:
                    multiplication();
                    break;

                case 4:
                    break;

                default:
                    System.out.println("INVALID INPUT. TRY AGAIN!!!");
                    break;
            }

        } while (option != 4);

        System.out.println("\n");

        final long duration = (System.nanoTime() - starttime)/1000000000;
        System.out.println("The system run for" + duration + " sec.");

    }

    public static void addition() {

        for(int i = 0; i < row; i++){
            for(int j = 0; j < column; j++){
                arr3[i][j] = arr1[i][j] + arr2[i][j];
            }
        }

        System.out.println("The result for addition is: ");
        for(int i = 0; i < row; i++){
            System.out.println("");
            for(int j = 0; j < column; j++){
                System.out.print(arr3[i][j]+"\t");
            }
        }

        System.out.println("");
    }

    public static void multiplication() {
        for(int i = 0; i < row; i++){
            for(int j = 0; j < column; j++){
                arr3[i][j] = 0;
                for(int k = 0; k < row; k++){
                    arr3[i][j] += arr1[i][k] * arr2[k][j];
                }
            }
        }

        System.out.println("The result for multiplication is: ");
        for(int i = 0; i < row; i++){
            System.out.println("");
            for(int j = 0; j < column; j++){
                System.out.print(arr3[i][j]+"\t");
            }
        }

        System.out.println("");
    }

    public static void subtraction() {
        for(int i = 0; i < row; i++){
            for(int j = 0; j < column; j++){
                arr3[i][j] = arr1[i][j] - arr2[i][j];
            }
        }

        System.out.println("The result for subtraction is: ");
        for(int i = 0; i < row; i++){
            System.out.println("");
            for(int j = 0; j < column; j++){
                System.out.print(arr3[i][j]+"\t");
            }
        }

        System.out.println("");
    }
}
